packages/external-compile/index.js

```diff
@@ -241,7 +241,7 @@ async function processTarget(target, cwd, logger) {
   }
 }
 
-const compile = callbackify(async function(options) {
+const compile = async function (options) {
```

Comment:
To be honest, I'm not sure what the purpose is of this extra compile function is. Do you have a clear idea of what its function is?

Reply:
As far as I know, the compile function is the "base" function - take some sources, some options, and output the compile result. So it's useful when you want something with less magic than `Compile.all` or `Compile.necessary`

Reply:
`compile-vyper` doesn't have the same signature either. Actually it is very different. Should we spend the time now to try and standardize that? Do you think that would be valuable? It might take a bit of doing... and I'm not sure `external-compile` would easily fit into a hole of that shape. Perhaps we could just leave it without a standard interface for now, what do you think?

Reply:
I think it's worth standardizing now, but would love to discuss why it's tricky!